src/lib/auth.ts

```diff
@@ -0,0 +1,18 @@
+import type { NextAuthOptions } from 'next-auth';
+import GoogleProvider from 'next-auth/providers/google';
+import { prisma } from './prisma';
+import { PrismaAdapter } from '@auth/prisma-adapter';
+
+export const authOptions: NextAuthOptions = {
+  providers: [
+    GoogleProvider({
+      clientId: process.env?.GOOGLE_CLIENT_ID ?? '',
```

Comment:
process.env should always exist, otherwise something is horribly wrong and should generate unexpected error

```suggestion
      clientId: process.env.GOOGLE_CLIENT_ID ?? '',
```

Reply:
This checks if GOOGLE_CLIENT_ID exists. If not undefined is returned and empty string is used instead. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

Reply:
Meant that if for some reason process.env is null/undef, we'll want an error. If that env property is undefined that nullish (??) check does the trick.

Edit. did some more googling and in order for that to happen someone has to explicitly set it as null so this is fine with or without that ? in the middle :D